### DISCUSSION TITLE
Gmaps navigation

### DISCUSSION THREAD

**Discussion by Krakuanen** [BimmerGestalt/AAIdrive#358] [Q&A]
Hi! I have compiled the Gmaps version and it seams to be working. The map is showing and I can search for a place. However, there are no turn by turn directions and no voice. I can only see the route as a line on the map. Have I missed something in the configuration ?

**Comment by hufman** [comment]
No, that is the expected experience. This limitation is intentional because it is against the Google Terms of Service to replicate the Google Navigation experience.
This Maps functionality is a Mapview and not actually Google Maps, and so some features are unable to be implemented.

**Reply by Krakuanen** [reply]
Ok, thanks. Maybe it would be possible to integrate some other map service without the limitations of Google?

**Reply by hufman** [reply]
Indeed, that is scheduled for 1.4 as part of #43

**Reply by Krakuanen** [reply]
Fantastic, looking forward to next release!

**Comment by patalmqx** [comment]
If you are able to create decent navigation with online traffic info, I could save ~60€/year for RTTI. It doesn't have to be google maps. But equivalently performance in terms of traffic info would be needed. I heared waze is also good? A few frames/second are not a real killer for me. You of course have boundary conditions..
This would be a real banger! Looking forward!!

**Reply by matthewblabey** [reply]
iirc waze is owned by Google so I would assume they will fall under the same terms of recreating Google maps in a car.

**Reply by hufman** [reply]
Waze is actually even harder, because it didn't have a public API to try to recreate its functionality. The plan is to build on top of Mapbox, which does provide a full SDK including navigation.
